src/util/dbid.h

```diff
@@ -0,0 +1,118 @@
+#ifndef DBID_H
+#define DBID_H
+
+
+#include <ostream>
+
+#include <QString>
+#include <QVariant>
+#include <QtDebug>
+
+#include "util/assert.h"
+
+
+// Base class for ID values of objects that are stored in the database.
+//
+// Implicit conversion from/to the native value type has been disabled
+// on purpose in favor of explicit conversion. The internal value type
+// is also hidden as an implementation detail.
+//
+// Although used as a base class this class has a non-virtual destructor,
+// because destruction is trivial and for maximum efficiency! Derived
+// classes may add additional behavior (= member functions), but should
+// not add any additional state (= member variables). Inheritance is
+// only needed for type-safety.
+class DbId {
+public:
+    DbId()
+        : m_value(kInvalidValue) {
+        DEBUG_ASSERT(!isValid());
+    }
+    explicit DbId(QVariant variant)
+        : m_value(valueOf(std::move(variant))) {
+        DEBUG_ASSERT(isValid() || (kInvalidValue == m_value));
+    }
```

Comment:
I think a alternative DbId(value_type value); constructor will help tao avoid implicit conversions to variant in com cases.